### ISSUE TITLE
Memory leak in libpostal libpostal_parse_address method

### ISSUE THREAD

**Issue by orividrisk** [bytedeco/javacpp-presets#1407]
Hi,
When using the libpostal 1.1-1.5.8 I experience a memory leak in my java process.
My code is iterating on a large number of addresses, and over time the java process memory is increasing until it crashes due to OOM.
I have monitored the jvm heap and non heap memory metrics and there is no issue there.
This leads me to the conclusion that the issue is with the native allocation of the returned address response, which is not deallocating the parsed addresses.
I did some debugging and found that the `libpostal_address_parser_response_t` object does not contain a deallocator, which if I understand correctly is in charge of deallocating the response object ( see in image).

The following is my code (I am using the default options provided by the library):
```

try (PointerScope pointerScope = new PointerScope();
             libpostal_address_parser_response_t response = libpostal_parse_address(addressStr, options)) {

            long count = response.num_components();
            Map<String, String> fields = new HashMap<>();

            for (int i = 0; i < count; i++) {
                fields.put(response.labels(i).getString(), response.components(i).getString());
            }

            libpostal_address_parser_response_destroy(response);
            return fields;
 }
```
![image](https://github.com/bytedeco/javacpp-presets/assets/91624812/e14a812b-846f-4253-8fc8-720e10eabda5)

I would very appreciate your prompt assistance with the issue as it is having a big impact on our production env.
Thank you and have a very nice day.

**Comment by saudet** [comment]
It's possible there is a leak in libpostal itself...

**Comment by orividrisk** [comment]
Thank you for the prompt response!

How can we verify if the issue is with `libpostal` or with the` javacpp-presets`?
When calling the `libpostal_parse_address` method which one is responsible for creating the deallocator object?

**Comment by saudet** [comment]
That's a normal C function call, so no deallocator gets created for
pointers returned.

**Comment by orividrisk** [comment]
Just to verify, does that mean that every method returning an object which inherits from `org.bytedeco.javacpp.Pointer` ( in this case: libpostal_address_parser_response_t) will not have a deallocator?
If that is the case is the expectation that libpostal will both return the result of the native method called, and deallocate the memory used to create the return type instance?

**Comment by saudet** [comment]
No, it just means that C functions returning a pointer don't have a
deallocator. I don't know how libpostal is supposed to be used. Please ask
upstream.

**Comment by orividrisk** [comment]
I understand, thank you.
I opened an issue in the upstream repo: https://github.com/openvenues/libpostal/issues/641
Waiting for their response
